rtt/Activity.cpp

```diff
@@ -163,7 +168,12 @@ namespace RTT
             return false;
         }
         mtimeout = true;
-        msg_cond.broadcast();
+        {
+#if defined(OROPKG_OS_XENOMAI) && (CONFIG_XENO_VERSION_MAJOR == 3)
+            os::MutexLock lock(msg_lock);
```

Comment:
Locking the mutex here is potentially critical because it creates a synchronization point between a thread that sends an operation and the one that has to process it. We had problems with priority inversions in the past even though the critical section is rather short.

Does Xenomai require to hold the mutex while broadcasting the condition variable?

Reply:
Ah, this has already been answered in https://github.com/orocos-toolchain/rtt/pull/224#issuecomment-336903027.